plugins/fresco-pd/fu-fresco-pd-firmware.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2020 Fresco Logic
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-fresco-pd-common.h"
+#include "fu-fresco-pd-firmware.h"
+
+struct _FuFrescoPdFirmware {
+	FuFirmwareClass		 parent_instance;
+	guint8			 customer_id;
+};
+
+G_DEFINE_TYPE (FuFrescoPdFirmware, fu_fresco_pd_firmware, FU_TYPE_FIRMWARE)
+
+guint8
+fu_fresco_pd_firmware_get_project_id (FuFrescoPdFirmware *self)
+{
+	return self->customer_id;
+}
+
+static void
+fu_fresco_pd_firmware_to_string (FuFirmware *firmware, guint idt, GString *str)
+{
+	FuFrescoPdFirmware *self = FU_FRESCO_PD_FIRMWARE (firmware);
+	fu_common_string_append_ku (str, idt, "CustomerID", self->customer_id);
+}
+
+static gboolean
+fu_fresco_pd_firmware_parse (FuFirmware *firmware,
+			     GBytes *fw,
+			     guint64 addr_start,
+			     guint64 addr_end,
+			     FwupdInstallFlags flags,
+			     GError **error)
+{
+	FuFrescoPdFirmware *self = FU_FRESCO_PD_FIRMWARE (firmware);
+	guint8 ver[4] = { 0x0 };
+	gsize bufsz = 0;
+	const guint8 *buf = g_bytes_get_data (fw, &bufsz);
+	g_autofree gchar *version = NULL;
+	g_autoptr(FuFirmwareImage) img = fu_firmware_image_new (fw);
+
+	/* read version block */
+	if (!fu_memcpy_safe (ver, sizeof(ver), 0x0,	/* dst */
+			     buf, bufsz, 0x1000,	/* src */
+			     sizeof(ver), error))
+		return FALSE;
+
+	/* project ID is always the 2nd byte */
```

Comment:
again using project and custom interchangably

Reply:
Fixed.